In [ ]:
import pandas as pd

df = pd.read_csv('result.csv')
df.head()

In [ ]:
df = df.filter(['id', 'text'])
df = df[df['text'].map(lambda x: len(x.split()) >= 3)]
df.head()

id                                               text
0  1389953924879110144  abis minum kopi perasaan mesti gelisah gatau k...
1  1389952303407329283          sumpah ya dari pagi rasanya gelisah terus
2  1389952181676122116  Catia hari sabtu.Presntasi Statin hari sabtu.....
3  1389951640610902020                                     tbtb gelisah 🤕
4  1389950790266023936  Memasuki jam jam rawan... Seliweran mulu di ga...

In [ ]:
len(df)

2751

In [ ]:
df = df.head(1000)
len(df)

1000

In [ ]:
df

id                                               text
0     1389953924879110144  abis minum kopi perasaan mesti gelisah gatau k...
1     1389952303407329283          sumpah ya dari pagi rasanya gelisah terus
2     1389952181676122116  Catia hari sabtu.Presntasi Statin hari sabtu.....
3     1389951640610902020                                     tbtb gelisah 🤕
4     1389950790266023936  Memasuki jam jam rawan... Seliweran mulu di ga...
...                   ...                                                ...
1019  1389076387684700162  Moots kalau berusaha tidur hadap kanan, hadap ...
1020  1389075326664286209  aku diem aja mbak nya cemas gmn kl aku gerak d...
1021  1389074928893267971  Tenang saja, semua akan baik baik saja. Tidak ...
1022  1389074441762611201  Tak sukanya orang beli baju berebut ni lepastu...
1023  1389074401883086850  Kalau pagi2 resah, gelisah, anxious itu tanda ...

[1000 rows x 2 columns]

Three annotators, so each of them will get 300 tweets + 100 shared tweets. In total:

$$3\times 300 + 100 = 1000$$

In [ ]:
split1, split2, split3, share = df[0:300], df[300:600], df[600:900], df[900:]

In [ ]:
split1.to_csv('split1.csv', index=False)
split2.to_csv('split2.csv', index=False)
split3.to_csv('split3.csv', index=False)
share.to_csv('share.csv', index=False)

# Creating DATD dataset

Merging back responses

In [ ]:
split1 = pd.read_csv('split1.csv')
split2 = pd.read_csv('split2.csv')
split3 = pd.read_csv('split3.csv')
share = pd.read_csv('share.csv')

In [ ]:
def load_response(response_name):
  res = []
  with open('{}.response'.format(response_name)) as f:
    res = [int(r.strip()) for r in f.readlines()]
  return res

In [ ]:
rsplit1, rsplit2, rsplit3 = load_response('split1'), load_response('split2'), load_response('split3')

In [ ]:
rshare1, rshare2, rshare3 = load_response('share-1'), load_response('share-2'), load_response('share-3')

In [ ]:
print(rshare1)
print(rshare2)
print(rshare3)

[0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 1, 0, 0, 0, 1, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 1, 0, 0, 0, 0, 0, 1, 0, 1, 1, 1, 0, 0, 1, 0, 0, 0, 1, 1, 1, 0, 0, 1, 1, 1, 0, 0, 0, 1, 1, 1, 0, 0, 0, 0, 1, 1, 0, 1, 0, 0, 0, 1, 0, 0, 0, 1, 0, 1, 1, 0, 0, 1, 0, 1, 1, 0, 0, 1, 0, 0, 0, 0, 0]
[1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 1, 0, 0, 1, 1, 1, 0, 0, 0, 0, 1, 0, 0, 0, 1, 1, 0, 0, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 1, 1, 0, 0, 1, 0, 0, 0, 1, 1, 1, 0, 1, 1, 0, 1, 1, 0, 0, 1, 1, 1, 0, 0, 0, 1, 0, 1, 0, 1, 0, 0, 0, 1, 0, 0, 1, 1, 0, 0, 1, 0, 0, 0, 0, 1, 1, 0, 1, 0, 0, 0, 1, 0, 0]
[1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 0, 0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 1, 0, 0, 0, 0, 0, 1, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1, 1, 0, 0, 1, 1, 0, 0, 0, 0, 1, 1, 1, 1, 0, 0, 0, 0, 1, 0, 1, 0, 0, 0, 1, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 1, 1, 0, 0, 1, 0, 0, 0, 0, 0]


In [ ]:
rshare = []

# three annotators, so tie is not possible
for i in range(len(rshare1)):
  zero = 0
  one = 0
  if rshare1[i] == 0:
    zero += 1
  else:
    one += 1
  
  if rshare2[i] == 0:
    zero += 1
  else:
    one += 1
  
  if rshare3[i] == 0:
    zero += 1
  else:
    one += 1
  
  if zero > one:
    rshare.append(0)
  else:
    rshare.append(1)

print(rshare)

[1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 1, 0, 0, 0, 1, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 1, 0, 0, 0, 0, 0, 1, 0, 1, 1, 1, 0, 0, 1, 0, 0, 0, 1, 1, 1, 0, 0, 1, 1, 1, 0, 0, 0, 1, 1, 1, 0, 0, 0, 0, 0, 1, 0, 1, 0, 0, 0, 1, 0, 0, 0, 1, 0, 1, 1, 0, 0, 0, 0, 1, 1, 0, 0, 1, 0, 0, 0, 0, 0]


In [ ]:
annotated_share = share.assign(label=rshare)

In [ ]:
annotated_split1 = split1.assign(label=rsplit1)
annotated_split2 = split2.assign(label=rsplit2)
annotated_split3 = split3.assign(label=rsplit3)

In [ ]:
annotated_dataset = pd.concat([annotated_split1, annotated_split2, annotated_split3], axis=0)

In [ ]:
annotated_dataset

id  ... label
0    1389953924879110144  ...     0
1    1389952303407329283  ...     1
2    1389952181676122116  ...     1
3    1389951640610902020  ...     0
4    1389950790266023936  ...     0
..                   ...  ...   ...
295  1389172826595160067  ...     0
296  1389169768003895298  ...     1
297  1389168219542286342  ...     0
298  1389168028416233472  ...     0
299  1389167572763901952  ...     0

[900 rows x 3 columns]

In [ ]:
from collections import Counter

label_counter = Counter(annotated_dataset['label'])
print(label_counter)

label_counter = Counter(annotated_share['label'])
print(label_counter)

Counter({0: 523, 1: 377})
Counter({0: 68, 1: 32})


Dropping `id` because it's not relevant for classification

In [ ]:
annotated_dataset = annotated_dataset.drop('id', axis=1)
annotated_share = annotated_share.drop('id', axis=1)

In [ ]:
print(annotated_dataset)

                                                  text  label
0    abis minum kopi perasaan mesti gelisah gatau k...      0
1            sumpah ya dari pagi rasanya gelisah terus      1
2    Catia hari sabtu.Presntasi Statin hari sabtu.....      1
3                                       tbtb gelisah 🤕      0
4    Memasuki jam jam rawan... Seliweran mulu di ga...      0
..                                                 ...    ...
295  Habis muncul bimbingan setelah menghilang berb...      0
296  Resah dan gelisah deh mimin, dah sebulan lewat...      1
297  Jangan membandingkan dirimu dengan orang lain....      0
298  Senin, 03 Mei 2020 / 21 Ramadhan 1442.Hari bai...      0
299       Gelisah, galau kalau seharian ga ada kabar..      0

[900 rows x 2 columns]


In [ ]:
print(annotated_share)

                                                 text  label
0   Aku merindukan tempatku yang tenang, tanpa ras...      1
1   Gelisah tanpa sebab. Terasa seperti punya kein...      1
2   Sekarang kita masih sama, suka bingung. Suka c...      0
3   aku mula lah gelisah bila dia outstation , kal...      0
4   Kasihan kamu, hidup dalam tekanan dan selalu d...      0
..                                                ...    ...
95  Moots kalau berusaha tidur hadap kanan, hadap ...      0
96  aku diem aja mbak nya cemas gmn kl aku gerak d...      0
97  Tenang saja, semua akan baik baik saja. Tidak ...      0
98  Tak sukanya orang beli baju berebut ni lepastu...      0
99  Kalau pagi2 resah, gelisah, anxious itu tanda ...      0

[100 rows x 2 columns]


In [ ]:
annotated_dataset.to_csv('datd_train.csv', index=False)
annotated_share.to_csv('datd_test.csv', index=False)

# Creating DATD+Rand dataset

In [ ]:
annotated_dataset[annotated_dataset['label'] == 1]

text  label
1            sumpah ya dari pagi rasanya gelisah terus      1
2    Catia hari sabtu.Presntasi Statin hari sabtu.....      1
11   Pernah gasi ngerasain tiap hari gelisah karena...      1
13   Aku makin gelisah, dan aku menangis sejadi jad...      1
14       ko aku dekdekan kenapa ya??aku gelisah banget      1
..                                                 ...    ...
280  Udah ngalihin fokus ke yang lain tapi tetep cemas      1
285  Kenapa ya cemas susah ilang? Deg2an panik itu ...      1
290  Kalo mau jujur sih gelisah sejak hari itu wala...      1
291  tiap malem overthinking, cemas, gelisah..tiap ...      1
296  Resah dan gelisah deh mimin, dah sebulan lewat...      1

[377 rows x 2 columns]

Based on the paper, DATD+Rand contained $10.5\%$ positive instances, so we will solve

$$\dfrac{377}{377+x} = 0.105$$

and we get $x = \lfloor 3213.48 \rfloor \approx 3213$

In [ ]:
random_df = pd.read_csv('random_dataset.csv')
random_df = random_df.filter(['text'])
random_df['label'] = 0
random_df.head(3213)

text  label
0     Gr gr mimpiin mantan crush jdi gamon gue ya Allah      0
1                                  wkwkwk apaansi lawak      0
2     Cuti2 pun standby depan pc 😅 sebab office tak ...      0
3            Alhamdulillah meskipun hanya numpang lewat      0
4     konsep teaser pernikahan pake lagu day6 - only...      0
...                                                 ...    ...
3208                        JA JA JA F POR EL JYP AHHHH      0
3209       Budak sekarang pasal tudung pun nak sensitif      0
3210  markah test semua dah keluar, cam okay lah sem...      0
3211  normalise hating someone by yourself bukan per...      0
3212       dilengangnya malam menuju minggu kau dimana~      0

[3213 rows x 2 columns]

In [ ]:
annotated_random = pd.concat([annotated_dataset[annotated_dataset['label'] == 1], random_df.head(3213)], axis=0)
annotated_random

text  label
1             sumpah ya dari pagi rasanya gelisah terus      1
2     Catia hari sabtu.Presntasi Statin hari sabtu.....      1
11    Pernah gasi ngerasain tiap hari gelisah karena...      1
14    Aku makin gelisah, dan aku menangis sejadi jad...      1
15        ko aku dekdekan kenapa ya??aku gelisah banget      1
...                                                 ...    ...
3208                        JA JA JA F POR EL JYP AHHHH      0
3209       Budak sekarang pasal tudung pun nak sensitif      0
3210  markah test semua dah keluar, cam okay lah sem...      0
3211  normalise hating someone by yourself bukan per...      0
3212       dilengangnya malam menuju minggu kau dimana~      0

[3590 rows x 2 columns]

In [ ]:
annotated_random.to_csv('datd_rand.csv', index=False)

# Adding more DATD entries

After making, training, and evaluating a model, we deemed that it is necessary to add more entries to the DATD dataset. Here, we will also export the rest of the DATD csv (entries beyond the 1000th), doing the same preprocessing as before (no entry with less than three tokens).

In [ ]:
df = pd.read_csv('result.csv')
df = df.filter(['id', 'text'])
df = df[df['text'].map(lambda x: len(x.split()) >= 3)]
df.head()

id                                               text
0  1389953924879110144  abis minum kopi perasaan mesti gelisah gatau k...
1  1389952303407329283          sumpah ya dari pagi rasanya gelisah terus
2  1389952181676122116  Catia hari sabtu.Presntasi Statin hari sabtu.....
3  1389951640610902020                                     tbtb gelisah 🤕
4  1389950790266023936  Memasuki jam jam rawan... Seliweran mulu di ga...

In [ ]:
df[1000:]

id                                               text
1024  1389073158800515074  G boleh raguin Tuhan, tapi kdng rasa cemas tu ...
1025  1389071564402958336  Jika kamu resah &amp; gelisah adukan saja kepa...
1026  1389070845910286338  Pernah menjadi yg paling cemas, sampai dipaksa...
1027  1389070688376418307            Resah resah gelisah,.Harap-harap cemas.
1028  1389070596219170823                      Dibuatnya gelisah pagi pagi 🙃
...                   ...                                                ...
2804  1387447629219438593                     gelisah dan panik yg luarbiasa
2805  1387446793407848451  Sedikit cemas tengok tracking status parcel ba...
2806  1387445987451445250  kaki aing balik lg kek wktu ituu, gelisah sblm...
2807  1387445659326812161  oke... aku lagi cemas... tiba tiba denger suar...
2808  1387445594394824704  3 malem loh gua gelisah, bengong trus tbtb nan...

[1751 rows x 2 columns]

In [ ]:
df[1000:].to_csv('additional.csv', index=False)

 # Merging additional DATD entries to the initial dataset

 Initial dataset has 900 train entries and 100 test entries. This is 90% : 10% split for train and test and we think 100 might be too few.

 The additional DATD dataset has 1751 more entires. We will somewhat merge this into train and test such that the ratio is about 80% : 20%. This means solving:

 $\dfrac{900+x}{1000+1751} = 0.8$

 and we get $x \approx 1300.8 \approx +1301$ for the train dataset and $1751-1301 = +450$ for the test dataset

 In the end, we will have 2201 train and 550 test

In [ ]:
additional_df = pd.read_csv('additional.csv')
additional_df.head()

id                                               text
0  1389073158800515074  G boleh raguin Tuhan, tapi kdng rasa cemas tu ...
1  1389071564402958336  Jika kamu resah &amp; gelisah adukan saja kepa...
2  1389070845910286338  Pernah menjadi yg paling cemas, sampai dipaksa...
3  1389070688376418307            Resah resah gelisah,.Harap-harap cemas.
4  1389070596219170823                      Dibuatnya gelisah pagi pagi 🙃

In [ ]:
radd1, radd2, radd3 = load_response('additional-1'), load_response('additional-2'), load_response('additional-3')

In [ ]:
print(radd1)
print(radd2)
print(radd3)

[1, 0, 0, 1, 1, 0, 1, 1, 1, 1, 1, 0, 0, 1, 1, 0, 0, 0, 0, 1, 1, 1, 1, 1, 0, 0, 0, 0, 0, 1, 1, 1, 0, 1, 0, 0, 0, 0, 0, 0, 1, 1, 1, 0, 1, 0, 0, 0, 1, 1, 0, 1, 0, 0, 1, 1, 1, 0, 1, 0, 0, 1, 0, 1, 1, 0, 1, 0, 0, 0, 0, 1, 1, 0, 1, 0, 0, 0, 1, 0, 0, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 1, 1, 0, 1, 0, 0, 1, 0, 0, 0, 1, 1, 1, 1, 0, 1, 1, 0, 0, 1, 0, 1, 1, 0, 0, 1, 1, 1, 0, 1, 1, 0, 1, 0, 1, 1, 0, 1, 0, 1, 1, 1, 1, 1, 1, 0, 1, 0, 0, 0, 0, 1, 1, 0, 0, 1, 1, 1, 1, 1, 1, 1, 0, 1, 0, 0, 0, 0, 1, 0, 0, 1, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 0, 1, 1, 0, 0, 1, 1, 1, 0, 1, 1, 0, 0, 1, 0, 1, 1, 0, 0, 0, 0, 0, 1, 1, 0, 0, 1, 1, 1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 0, 1, 0, 1, 0, 0, 1, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0, 0, 0, 0, 1, 1, 0, 1, 0, 0, 1, 1, 0, 0, 1, 1, 0, 1, 0, 0, 1, 0, 0, 0, 1, 0, 1, 1, 0, 1, 1, 0, 0, 0, 1, 0, 1, 0, 1, 1, 1, 0, 0, 1, 1, 0, 1, 0, 0, 1, 1, 1, 1, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 1, 0, 0, 1, 1, 1, 0, 1, 1, 0, 1, 0, 0, 1, 1, 0, 0, 0, 1, 1, 1, 1, 0, 1, 0, 0, 0, 1, 1, 0, 0, 

In [ ]:
radditional = []

# three annotators, so tie is not possible
for i in range(len(radd1)):
  zero = 0
  one = 0
  if radd1[i] == 0:
    zero += 1
  else:
    one += 1
  
  if radd2[i] == 0:
    zero += 1
  else:
    one += 1
  
  if radd3[i] == 0:
    zero += 1
  else:
    one += 1
  
  if zero > one:
    radditional.append(0)
  else:
    radditional.append(1)

print(radditional)

[1, 0, 0, 1, 0, 0, 1, 0, 1, 1, 1, 0, 0, 1, 1, 0, 0, 0, 0, 0, 1, 0, 1, 1, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 0, 0, 0, 0, 0, 0, 1, 1, 0, 1, 0, 0, 0, 0, 1, 0, 0, 1, 0, 0, 0, 1, 0, 0, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 1, 1, 0, 1, 0, 0, 1, 0, 0, 0, 0, 1, 1, 0, 0, 1, 1, 0, 0, 1, 0, 1, 1, 0, 0, 1, 0, 0, 0, 1, 0, 0, 1, 0, 1, 0, 0, 1, 0, 1, 1, 1, 1, 1, 1, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 1, 1, 1, 1, 1, 1, 0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 1, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 0, 1, 1, 0, 0, 0, 0, 1, 0, 1, 1, 0, 0, 1, 0, 1, 1, 0, 0, 0, 0, 0, 0, 1, 0, 0, 1, 0, 1, 1, 0, 1, 1, 1, 1, 0, 1, 1, 0, 1, 0, 1, 0, 0, 1, 0, 0, 0, 1, 1, 0, 1, 1, 1, 1, 0, 0, 0, 0, 1, 1, 1, 0, 1, 0, 0, 1, 1, 0, 0, 0, 1, 0, 1, 0, 0, 1, 0, 0, 0, 1, 0, 1, 1, 0, 0, 1, 0, 0, 0, 1, 0, 1, 0, 1, 0, 0, 0, 0, 0, 1, 0, 1, 0, 0, 1, 1, 0, 1, 0, 0, 1, 1, 0, 0, 0, 1, 0, 1, 1, 0, 1, 0, 0, 1, 1, 1, 0, 1, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 1, 0, 1, 0, 0, 0, 1, 1, 0, 0, 

In [ ]:
annotated_additional = additional_df.assign(label=radditional).drop('id', axis=1)
annotated_additional = annotated_additional.sample(frac=1, random_state=15).reset_index(drop=True)
annotated_additional

text  label
0                         Gelisah tanpa sebab. Ko bisa?      1
1     kawan aku chat aku, kata ada berita buruk. but...      0
2     Cemas terus setiap bulannya, apalagi sedang be...      1
3                           kok bawaannya gelisah terus      1
4                                Gelisah gini woyyyiii😊      1
...                                                 ...    ...
1746                resah gelisah nak another 2 album 😔      0
1747  Crypto membuatku gelisah gabisa tidur, nanti m...      0
1748                    sedikit cemas banyak gila nya 😭      0
1749  Gelisah ga karuan, tapi cuma bisa di pendem. Y...      1
1750  dunia bener2 galucu kalo nyokap sakit. apa2 be...      1

[1751 rows x 2 columns]

In [ ]:
current_train = pd.read_csv('datd_train.csv')
current_test = pd.read_csv('datd_test.csv')

In [ ]:
current_train

text  label
0    abis minum kopi perasaan mesti gelisah gatau k...      0
1            sumpah ya dari pagi rasanya gelisah terus      1
2    Catia hari sabtu.Presntasi Statin hari sabtu.....      1
3                                       tbtb gelisah 🤕      0
4    Memasuki jam jam rawan... Seliweran mulu di ga...      0
..                                                 ...    ...
895  Habis muncul bimbingan setelah menghilang berb...      0
896  Resah dan gelisah deh mimin, dah sebulan lewat...      1
897  Jangan membandingkan dirimu dengan orang lain....      0
898  Senin, 03 Mei 2020 / 21 Ramadhan 1442.Hari bai...      0
899       Gelisah, galau kalau seharian ga ada kabar..      0

[900 rows x 2 columns]

In [ ]:
train_final = pd.concat([current_train, annotated_additional[:1301]], axis=0)
train_final = train_final.sample(frac=1, random_state=15).reset_index(drop=True)
train_final

text  label
0     oh pantesan tadi pada rame, ternyata monek mau...      0
1          Semakin bertambah usia, semakin cemas hidup.      0
2                                    gelisah bgt astaga      1
3     Udah jangan terlalu cemas sikapku tak berubah ...      0
4     Giliran Aldebaran diambang kematian...Semua ba...      0
...                                                 ...    ...
2196  hey please boleh minta kata kata positif or se...      1
2197  ekpektasimu mematahkanmu.harapmu tak sesuai ke...      0
2198  Rasa takut , cemas dan tiba tiba deg deg an te...      0
2199  Rakyat merasa tenang ada FPI, malah gelisah ad...      0
2200      Lagi lagi saya cemas sama tlp yg masuk ke hp.      0

[2201 rows x 2 columns]

In [ ]:
test_final = pd.concat([current_test, annotated_additional[1301:]], axis=0)
test_final = test_final.sample(frac=1, random_state=15).reset_index(drop=True)
test_final

text  label
0    ngga enak bgt akhir2 ini rasanya, sering cemas...      1
1                      Sedikit khawatir dan gelisah. 😓      1
2    Ngelakuin kesalahan yang sama. Bodoh bgt. Udah...      1
3          Pantesan gelisah bae, lupa lgi ga ada duit🤭      0
4    Semoga Mei disertai kabar-kabar baik. Yang luk...      0
..                                                 ...    ...
545                     gelisah, padahal gitu doang bd      1
546  Kekhawatiran berlebih akan membuat stress, ngg...      0
547  otak gue kalo malem gini suka kemana mana hera...      1
548  Kalo lg gelisah beban banyak pikiran trs langs...      0
549  Astagfirullah....Astagfirullah....Astagfirulla...      0

[550 rows x 2 columns]

In [ ]:
train_final.to_csv('datd_train_additional.csv', index=False)
test_final.to_csv('datd_test_additional.csv', index=False)

Now we do the same thing for DATD+Rand

In [ ]:
train_ds = pd.read_csv('datd_train_additional.csv')
train_ds[train_ds['label'] == 1]

text  label
2                                    gelisah bgt astaga      1
5     tidur susah, pikiran kacau, hati gelisah. Semu...      1
6     Serba salah, mundur sayang, maju ragu. Diem di...      1
7     Udah ngalihin fokus ke yang lain tapi tetep cemas      1
8     Resah dan gelisah deh mimin, dah sebulan lewat...      1
...                                                 ...    ...
2187                 gua cemas bgt tp ga tau karena apa      1
2191  Gak pernah kelewat buat nguatin diri sendiri k...      1
2192  Ini gelisah bngttt dri kemarin mikirkan bntr l...      1
2194                                  Resah dan gelisah      1
2196  hey please boleh minta kata kata positif or se...      1

[733 rows x 2 columns]

$$\dfrac{733}{733+x} = 0.105$$

$x = \lfloor 6247.95 \rfloor \approx 6247$

In [ ]:
random_additional = pd.concat([train_ds[train_ds['label'] == 1], random_df.head(6247)], axis=0)
random_additional

text  label
2                                    gelisah bgt astaga      1
5     tidur susah, pikiran kacau, hati gelisah. Semu...      1
6     Serba salah, mundur sayang, maju ragu. Diem di...      1
7     Udah ngalihin fokus ke yang lain tapi tetep cemas      1
8     Resah dan gelisah deh mimin, dah sebulan lewat...      1
...                                                 ...    ...
6242         aku tidak bisa melupakan mu #SpiritHadrian      0
6243  【定期☆】..DJ DAIKI! DJ DAIKI! DJ DAIKI!.DJ DAIKI!...      0
6244                                       Dai5y! &lt;3      0
6245  capek badut tp teteup semoga malem ini kalo ga...      0
6246                                 MW SUBI WN LA VACA      0

[6980 rows x 2 columns]

In [ ]:
random_additional.to_csv('datd_rand_additional.csv', index=False)